benchexec/tools/metaval.py

```diff
@@ -77,12 +78,19 @@ def determine_result(self, returncode, returnsignal, output, isTimeout):
         ):
             return "METAVAL ERROR"
         verifierName = self.PATH_TO_TOOL_MAP[verifierDir]
-        with self._in_tool_directory(verifierName):
-            return self.wrappedTools[verifierName].determine_result(
-                returncode, returnsignal, output, isTimeout
-            )
 
-    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
+        tool = self.wrappedTools[verifierName]
+        if isinstance(tool, BaseTool2):
+            return tool.determine_result(returncode, returnsignal, output, isTimeout)
+        elif isinstance(tool, BaseTool):
```

Comment:
Should this be removed as well? I would assume that the import of `BaseTool` as well as `_in_tool_directory` can be removed completely now.

Reply:
Sure, done!